### ISSUE TITLE
i don't like the workflow of creating new datasource

### ISSUE THREAD

**Issue by nagy-nabil** [nilefy/nilefy#173]
## current workflow

the current implementation is a replica of Tooljet workflow
- when the user clicks create: only request data source name
	
	![image](https://github.com/z-grad-pr-sh/frontend/assets/65990275/bf3504cc-2840-4220-96f4-80fd6d4ac369)

- if the data source was created successfully(server return 201): redirect the user to the page where they can really configure the data source
	
	![image](https://github.com/z-grad-pr-sh/frontend/assets/65990275/411c125d-0266-4afd-81f6-77213a3898a7)

## why i don't think this good workflow?

any data source will have some required configuration that it cannot operate without, and i think when a user creates a data source they will probably use it so they need to enter those required fields for the data source to be able to work,
so basically  now the user creates the data source in two steps and the second step(configuration) is optional while really this step is required

## A clear and concise description of what you want to happen.

make the configuration step part of the creation workflow

**Describe alternatives you've considered**

check how other platforms handle this case, if i remember correctly both illabuilder and appSmith have the configuration as part of the workflow, not a separate step
